Introduce JRL CMake Modules v2

ahoarau opened the pull request:
In this PR we introduce the `JRL CMake Modules v2`, extracted from the repo: https://github.com/ahoarau/jrl-cmakemodules-v2

The integration on the user-side can be done as follows https://github.com/ahoarau/nanoeigenpy/blob/db692b1dbe3fff76688550048fd3004ecbd1a2d7/cmake/get-jrl-cmakemodules.cmake

```cmake
find_package(jrl-cmakemodules 1.2.0 CONFIG REQUIRED)
```

* v1.2.0 will integrate the v2 folder, containing the new API.
* Either set an env variable `JRL_CMAKEMODULES_SOURCE_DIR` pointing to the jrl repo source (example `$env:JRL_CMAKEMODULES_SOURCE_DIR="C:/a/jrl-cmakemodules"`
* We do **not** intend to use both `v1` and `v2` in projects. 

TODO:
- [x] Port the main modules
- [x] Test the integration in client repo (proxsuite, eigenpy, nanoeigenpy, coal)
- [x] Figure out why gersemi still needs to re-format the new modules
- [x] Add `COMPATIBLITY_OPTION` for `jrl_cmake_dependent_option()`
- [x] Port tests from original repo
- [x] Port CI from original repo
- [x] Rewrite `release.cmake` in python
- [x] Port pypa metadata

jorisv commented on v2/jrl-cmakemodules-v2/templates/__init__.py.in:
```diff
@@ -0,0 +1,37 @@
+    mgr.safe_add_dll_directory(os.path.join(module_path, '..', '..', '..', 'bin'))
+    dll_dirs = @__DLL_DIRS__@
+    for dll_dir in dll_dirs:
+        mgr.safe_add_dll_directory(os.path.join(module_path, dll_dir))
```
I think `DllDirectoryManager.__del__` will never be called here.

mgr will never be delete when we import `@__MODULE_NAME__@`.

You can check it with `toto.py`:

```python
class Toto:
    def __init__(self):
        print("toto.__init__")

    def __del__(self):
        print("toto.__del__")


t = Toto()
```

If you `import toto` Only enter will be called.

You need to use contextlib to handle it well.

jorisv commented on CMakeLists.txt:
```diff
@@ -1,5 +1,10 @@
 cmake_minimum_required(VERSION 3.22)
 
+if(JRL_CMAKEMODULES_USE_V2)
```
`JRL_CMAKEMODULES_USE_V2` should be an option

ahoarau replied:
done ✅

jorisv commented on v2/jrl-cmakemodules-v2/modules/jrl.cmake:
```diff
@@ -0,0 +1,1788 @@
+    endforeach()
+endfunction()
+
+function(jrl_target_set_output_directory target_name)
```
There is no documentation. Is this function still useful ?

jorisv commented on v2/jrl-cmakemodules-v2/modules/jrl.cmake:
```diff
@@ -0,0 +1,1788 @@
+endfunction()
+
+# Description: Treat all warnings as errors for a targets (/WX for MSVC, -Werror for GCC/Clang)
+# Can be disabled by on the cmake cli with --compile-no-warning-as-error
```
```suggestion
# Can be disabled on the cmake cli with --compile-no-warning-as-error
```

jorisv commented on v2/jrl-cmakemodules-v2/modules/jrl.cmake:
```diff
@@ -0,0 +1,1788 @@
+    set(${OUTPUT_VAR} "${CLEAN}" PARENT_SCOPE)
+endfunction()
+
+function(jrl_target_generate_header target_name visibility)
```
Add some documentation

jorisv commented on v2/jrl-cmakemodules-v2/modules/jrl.cmake:
```diff
@@ -0,0 +1,1788 @@
+    cmake_parse_arguments(arg "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+    jrl_check_var_defined(PROJECT_NAME)
+    jrl_check_var_defined(CMAKE_CURRENT_BINARY_DIR)
```
Is this one mandatory ?

jorisv commented on v2/jrl-cmakemodules-v2/modules/jrl.cmake:
```diff
@@ -0,0 +1,1788 @@
+    )
+endfunction()
+
+function(jrl_target_generate_deprecated_header target_name visibility)
```
Add some documentation

jorisv commented on v2/jrl-cmakemodules-v2/modules/jrl.cmake:
```diff
@@ -0,0 +1,1788 @@
+# cf: https://cmake.org/cmake/help/latest/command/target_sources.html#file-sets
+function(jrl_target_headers target visibility)
+    set(options)
+    set(oneValueArgs GENERATED_DIR)
```
GENERATED_DIR is never used

jorisv commented on v2/jrl-cmakemodules-v2/modules/jrl.cmake:
```diff
@@ -0,0 +1,1788 @@
+    set_property(GLOBAL PROPERTY _jrl_${PROJECT_NAME}_option_names ${option_name} APPEND)
+endfunction()
+
+function(
```
Add some documentation

jorisv commented on v2/jrl-cmakemodules-v2/modules/jrl.cmake:
```diff
@@ -0,0 +1,1788 @@
+    # If you install nanobind with pip, it will include tsl-robin-map in <nanobind>/ext/robin_map
+    # On macOS, brew install nanobind will not include tsl-robin-map, we need to install it via: brew install robin-map
+    # Naturally, find_package(nanobind CONFIG REQUIRED) will succeed (nanobind_FOUND -> True), but the tsl-robin-map dependency will be missing, causing build errors.
+    # So let's check if the headers are available, otherwise require tsl-robin-map explicitly.
```
In this [document](https://nanobind.readthedocs.io/en/latest/api_cmake.html#submodule-dependencies) they speak about NB_USE_SUBMODULE_DEPS variable to set if we use nanobind as a system dependency.

jorisv commented on v2/jrl-cmakemodules-v2/modules/jrl.cmake:
```diff
@@ -0,0 +1,1788 @@
+    endif()
+endfunction()
+
+function(jrl_python_generate_init_py name)
```
Add some documentation

jorisv commented on v2/jrl-cmakemodules-v2/CMakeLists.txt:
```diff
@@ -0,0 +1,33 @@
+if(NOT PROJECT_IS_TOP_LEVEL)
+    return()
+endif()
+
```
I don't get it. We will return if included from the main CMakeLists.txt. Then we will not generate the configuration file.

ahoarau commented on v2/jrl-cmakemodules-v2/find-modules/FindAccelerate.cmake:
Done ✅

jorisv commented on v2/modules/jrl.cmake:
```diff
@@ -0,0 +1,1957 @@
+# Works on all platforms
+function(jrl_configure_default_install_dirs)
+    include(GNUInstallDirs)
+endfunction()
```
```suggestion
macro(jrl_configure_default_install_dirs)
    include(GNUInstallDirs)
endmacro()
```

Some variables like `${CMAKE_INSTALL_FULL_BINDIR}` are not defined if the include is inside a function.

jorisv replied:
No use case in mind. If we know that all non cached variable are unused this can be kept as a function.

jorisv commented on v2/modules/jrl.cmake:
```diff
@@ -0,0 +1,1957 @@
+# Compute the installation directory for Python bindings.
+#  * If ${PROJECT_NAME}_PYTHON_INSTALL_DIR is defined, its value is used.
+#  * Otherwise, if running inside a Conda environment on Windows, an
+#    absolute path to `sysconfig.get_path('purelib')` is returned.
```
As we already talked, I'm not a huge fan of handling this special case. The `${PROJECT_NAME}_PYTHON_INSTALL_DIR` variable is made for this purpose.

ahoarau replied:
I tried both options to see which one feels more natural. In the end, this function is here to handle the different cases of a python install dir, so it makes sense to take the conda+windows special/unique/annoying case here. All the other cases are fine with the default behaviour, so I think it's wise to keep it.

The `${PROJECT_NAME}_PYTHON_INSTALL_DIR` is still useful as it can be used to override this behavior.

ahoarau committed:
rename to cmake_uninstall.cmake to matche the cmake_install.cmake

ahoarau committed:
remove the legacy IMMEDIATE keyword in configure_file
https://github.com/Kitware/CMake/blob/master/Source/cmConfigureFileCommand.cxx#L82-L83C6

ahoarau committed:
rename jrl_setup_uninstall_target to jrl_configure_uninstall_target for uniformity

ahoarau committed:
jrl_python_get_interpreter: make sure the path is converted. into a cmake path list

ahoarau committed:
enable policy CMP0177: add normalization for install() DESTINATION paths to handle Windows backslash issues

ahoarau committed:
normalize paths when using cmake_path

ahoarau committed:
Revert "enable policy CMP0177: add normalization for install() DESTINATION paths to handle Windows backslash issues"

This reverts commit ae6e209de8a692afcd4750e03d01f237bd3b236d.

ahoarau committed:
Add function to retrieve git commit hash and display in banner

ahoarau committed:
convert user defined <pkg>_PYTHON_INSTALL_DIR to cmake_path_list and normalize

ahoarau commented on v2/modules/jrl.cmake:
```diff
@@ -0,0 +1,1788 @@
+            "Python_EXECUTABLE not defined.
+        Please use jrl_find_package(Python REQUIRED COMPONENT Interpreter)"
+        )
+    endif()
```
Let's use the target version everywhere to prevent any env pollution from the exterior (like Conda which seems to provide `Python_EXECUTABLE` in the command line).

Added a `jrl_python_get_interpreter`

ahoarau committed:
run updated gersemi formatter

ahoarau committed:
merge BoostPython.cmake into main jrl module

ahoarau committed:
merge check python module name into main module

arntanguy reviewed (commented):
Just gave it a try with SpaceVecAlg: https://github.com/jrl-umi3218/SpaceVecAlg/pull/67
There seems to be an issue with BUILDSYSTEM_TARGETS, see my comment inline.

Besides that it worked just fine. I'll keep exploring during the week.

ahoarau committed:
add jr_log internal functions

ahoarau committed:
remove the buildsystem target filtering
its difficult to get them all correctly here + we dont support generator expressions. It's fine to remove it as it's just meant as a security/info for users

ahoarau committed:
ignore .DS_Store

ahoarau committed:
add jrl_target_generate_config_header documentation (new format)

ahoarau committed:
use _jrl_log in jrl_print_dependencies_summary

ahoarau committed:
fix pytest on ubuntu 22.04
its named pytest-3 and prints version in stderr. go figure.

nim65s commented on v2/find-modules/FindScipy.cmake:
```diff
@@ -0,0 +1,29 @@
+# Copyright 2025-2026 Inria
+
+if(NOT DEFINED Python_EXECUTABLE)
```
harmonize with FindPytest

nim65s commented on v2/modules/jrl.cmake:
```diff
@@ -0,0 +1,2276 @@
+# Usage: jrl_check_var_defined(<var> [<message>])
+# Example: jrl_check_var_defined(MY_VAR "MY_VAR must be set to build this project")
+# Example: jrl_check_var_defined(MY_VAR) # Will print "MY_VAR is not defined."
+function(jrl_check_var_defined var)
```
`assert` would be more explicit than `check`
same below.

ahoarau replied:
Fair point.
The first version was `require_var_defined()`.
Maybe I should set them private to limit the amount of public functions of this lib. What do you think ?

nim65s commented on v2/modules/jrl.cmake:
```diff
@@ -0,0 +1,2276 @@
+    jrl_configure_default_build_type(Release)
+    jrl_configure_default_binary_dirs()
+    jrl_configure_default_install_dirs()
+    jrl_configure_default_install_prefix(${CMAKE_BINARY_DIR}/install)
```
I never thought about this. At first sight, this looks like a very good idea to me, thanks :)

nim65s commented on v2/modules/jrl.cmake:
```diff
@@ -0,0 +1,2276 @@
+#
+# Notes:
+#   * This macro is a convenience wrapper and does not change the fundamental semantics of find_package.
+#   * Prefer using REQUIRED to ensure missing packages are caught early. Don't react to missing packages manually.
```
> Don't react to missing packages manually

I don't understand this.

If I'm a package author, and I want to add an optional dependency, am I not supposed to react to that optional dependency missing manually ?

nim65s commented on v2/modules/jrl.cmake:
```diff
@@ -0,0 +1,2276 @@
+#   version        - Optional version requirement forwarded to find_package.
+#   COMPONENTS ... - Optional list of components forwarded to find_package.
+#   MODULE_PATH    - Optional path to a custom Find<PackageName>.cmake module file.
+#                    If specified, only this path is used, and the file must be at <path_to_find_module>/Find<PackageName>.cmake
```
> only this path is used

I don't see that is the case in `list(APPEND CMAKE_MODULE_PATH ${module_dir})`

nim65s commented on v2/modules/jrl.cmake:
```diff
@@ -0,0 +1,2276 @@
+        "Mode keyword missing in jrl_cmake_print_properties() call, there must be exactly one of ${_mode_names}"
+    )
+
+    cmake_parse_arguments(CPP "${options}" "${oneValueArgs}" "${cpp_multiValueArgs}" ${ARGN})
```
the function above use "arg". 
I think we should not have different naming scheme for the same things

ahoarau replied:
adapted from https://github.com/Kitware/CMake/blob/master/Modules/CMakePrintHelpers.cmake
Added a comment for the ref.

Maybe I'll rewrite this function later. It's overly complex for my use-case

nim65s commented on v2/modules/jrl.cmake:
```diff
@@ -0,0 +1,2276 @@
+        set(FILENAME ${PROJECT_NAME}-dependencies.cmake)
+    endif()
+
+    # TODO: filter the buildsystems targets of the INTERFACE_LINK_LIBRARIES.
```
what are buildsystems targets ?

ahoarau replied:
https://cmake.org/cmake/help/latest/prop_dir/BUILDSYSTEM_TARGETS.html

All targets (library, exec) of the current build tree, not imported targets.

nim65s commented on v2/modules/jrl.cmake:
````diff
@@ -0,0 +1,2276 @@
+# jrl_python_get_interpreter(python_interpreter)
+# execute_process(COMMAND ${python_interpreter} -c "print('Hello from Python!')")
+# ```
+function(jrl_python_get_interpreter output_var)
````
why do we need this ? and if we need this, why do we use Python_EXECUTABLE is other places in this file ?

ahoarau replied:
Update: we now use `jrl_python_get_interpreter` everywhere.

nim65s commented on v2/modules/jrl.cmake:
```diff
@@ -0,0 +1,2276 @@
+
+    execute_process(
+        COMMAND
+            ${python} -c
```
using this module CLI instead of python API would be more clear

nim65s commented on v2/templates/generate-dependencies.cmake.in:
```diff
@@ -0,0 +1,99 @@
+    message(FATAL_ERROR "package_dependencies_json_content variable is not defined in @IMPORTED_LIBRARIES_TO_EXPORT_FILE@.")
+endif()
+
+if(package_dependencies_json_content)
```
remove this if

nim65s reviewed (commented):
That's a lot of work 😅 

I did not look into external-modules yet, but here are some comments as a starting point.

ahoarau committed:
remove FindScipy we now have jrl_check_python_module

ahoarau committed:
jrl_check_python_module: get module version if available

ahoarau committed:
generate-dependencies.cmake.in: add comments to clarify the function conditions

ahoarau committed:
use jrl_check_file_exists when possible

ahoarau committed:
update jrl_find_package doc

ahoarau commented on v2/modules/jrl.cmake:
```diff
@@ -0,0 +1,2276 @@
+# Setup the default options for a project (opinionated defaults)
+# Usage : jrl_configure_defaults()
+function(jrl_configure_defaults)
+    jrl_configure_default_build_type(Release)
```
That was my first proposal, but in the end I synced with what we had in pixi files (Release)

ahoarau replied:
Seen with joris that has a strong opinion on keeping Release, lets close this one.

ahoarau commented on v2/modules/jrl.cmake:
```diff
@@ -0,0 +1,2276 @@
+    string(REPLACE "." ";" version_parts ${LIBRARY_VERSION})
+    list(GET version_parts 0 LIBRARY_VERSION_MAJOR)
+    list(GET version_parts 1 LIBRARY_VERSION_MINOR)
+    list(GET version_parts 2 LIBRARY_VERSION_PATCH)
```
Indeed. What about checking if they exists and put 0 if not available ?

ahoarau commented on v2/modules/jrl.cmake:
```diff
@@ -0,0 +1,2276 @@
+# This function is used to determine which module file was used by jrl_find_package.
+# Usage: jrl_search_package_module_file(<package_name> <output_filepath>)
+# Example: jrl_search_package_module_file(Eigen module_file)
+function(jrl_search_package_module_file package_name output_filepath)
```
still shorter than `jrl_target_generate_warnings_header`  !

ahoarau committed:
update jrl_find_package doc on MODULE_PATH

ahoarau committed:
unset variables at the end and add comments

ahoarau committed:
jrl_find_package: reduce verbosity of variables msg

ahoarau committed:
cmake_print_properties: added ref

ahoarau committed:
jrl_export_dependencies: edit doc

ahoarau commented on v2/modules/jrl.cmake:
````diff
@@ -0,0 +1,2276 @@
+# )
+# ```
+# The generated __init__.py will call the os.add_dll_directory(<relative_path/to/coal.dll>)
+function(jrl_python_generate_init_py name)
````
Yes. but i'd rather generate the same exact file on different platforms. Easier debugging 👍

ahoarau committed:
jrl_boostpy_add_stubs: rename OUTPUT to OUTPUT_PATH and compute the stub name

ahoarau committed:
add jrl_python_relative_site_packages (and absolute)
* refactor jrl_python_compute_install_dir
* add documentation

ahoarau committed:
add jrl_generate_ros2_package_files

ahoarau committed:
jrl_generate_ros2_package_files: python generation on confition INSTALL_PYTHON_PACKAGE_FILES

ahoarau committed:
jrl_generate_ros2_package_files: check DEFINED, not value

ahoarau committed:
jrl_generate_ros2_package_files: condition cpp files gen

ahoarau committed:
use CMAKE_CXX_COMPILER_FRONTEND_VARIANT if available

ahoarau committed:
jrl_export_dependencies: require CMAKE_INSTALL_LIBDIR

ahoarau committed:
generate-dependencies: handle absolute and relative install destinations

ahoarau committed:
jrl_target_install_headers: handle abs/rel install destination and abs/rel CMAKE_INSTALL_INCLUDEDIR

ahoarau committed:
reformat doc as a uniform markdown style

fixes on function signatures and docs

ahoarau committed:
remove unused jrl_contains_generator_expressions

ahoarau committed:
make _jrl_make_valid_c_identifier internal

ahoarau committed:
make _jrl_export_dependencies intenal

ahoarau committed:
uniformize the variables names

ahoarau committed:
internalize check functions

ahoarau committed:
add _jrl_normalize_version

ahoarau committed:
jrl_target_headers: remove unused check

ahoarau committed:
templates: prepend variables with JRL

ahoarau committed:
refactor header generations functions

ahoarau committed:
refactor header generations functions

ahoarau committed:
remove unused external modules

ahoarau committed:
remove unused external modules

ahoarau committed:
add include_guard(GLOBAL)

ahoarau committed:
add include_guard(GLOBAL)

ahoarau committed:
_jrl_top_dir: remove trailing slash if any

ahoarau committed:
_jrl_top_dir: remove trailing slash if any

ahoarau committed:
add _jrl_docs_dir

ahoarau committed:
add _jrl_docs_dir

ahoarau committed:
ignore api.md

ahoarau committed:
ignore api.md

ahoarau committed:
refactor _jrl_generate_api_doc

ahoarau committed:
refactor _jrl_generate_api_doc

ahoarau committed:
docs: update api.md

ahoarau committed:
docs: update api.md

ahoarau committed:
update documentation format

ahoarau committed:
update documentation format

ahoarau committed:
remove range on cmake min version

ahoarau committed:
remove range on cmake min version

ahoarau committed:
add option to build doc

ahoarau committed:
add option to build doc

ahoarau committed:
add initial README

ahoarau committed:
add initial README

ahoarau committed:
make jrl_search_package_module_file private

ahoarau committed:
make jrl_search_package_module_file private

ahoarau committed:
remove jrl_cmake_print_properties

ahoarau committed:
remove jrl_cmake_print_properties

ahoarau committed:
update documentation

ahoarau committed:
update documentation

ahoarau committed:
filter out internal functions in doc

ahoarau committed:
filter out internal functions in doc

ahoarau committed:
gitignore: ignore ruff_cache

ahoarau committed:
gitignore: ignore ruff_cache

ahoarau committed:
jrl_add_export_component: remove comment and TODO

ahoarau committed:
jrl_add_export_component: remove comment and TODO

ahoarau committed:
jrl_print_dependencies_summary: fix padding

ahoarau committed:
jrl_print_dependencies_summary: fix padding

ahoarau committed:
jrl_python_compute_install_dir: allow to use env var. Improve doc.

ahoarau committed:
jrl_python_compute_install_dir: allow to use env var. Improve doc.

ahoarau committed:
doc: update api doc

ahoarau committed:
doc: update api doc

ahoarau committed:
jrl_find_nanobind: use interpreter and improve doc

ahoarau committed:
jrl_find_nanobind: use interpreter and improve doc

ahoarau committed:
refactor jrl_find_package with jrl_export_dependency

ahoarau committed:
refactor jrl_find_package with jrl_export_dependency

ahoarau committed:
_jrl_generate_api_doc: fix nested semicolons case

ahoarau committed:
_jrl_generate_api_doc: fix nested semicolons case

ahoarau committed:
doc: update api doc

ahoarau committed:
doc: update api doc

ahoarau committed:
remove _jrl_make_valid_c_identifier, use built in string(MAKE_C_IDENTIFIER <string> <output_variable>)

ahoarau committed:
remove _jrl_make_valid_c_identifier, use built in string(MAKE_C_IDENTIFIER <string> <output_variable>)

ahoarau committed:
rename the option to JRL_GENERATE_API_DOC

ahoarau committed:
rename the option to JRL_GENERATE_API_DOC

ahoarau committed:
refactor jrl_option, remove jrl_c;ake_dependant_option, and add jrl_legacy_option

ahoarau committed:
refactor jrl_option, remove jrl_c;ake_dependant_option, and add jrl_legacy_option

ahoarau committed:
allow to disable export via project-based option

ahoarau committed:
allow to disable export via project-based option

ahoarau committed:
uppercase the python installdir bypass option

ahoarau committed:
uppercase the python installdir bypass option

ahoarau committed:
update api documentation

ahoarau committed:
update api documentation

ahoarau committed:
add first draft of testing framework

ahoarau committed:
add first draft of testing framework

ahoarau committed:
always set the option in CACHE

ahoarau committed:
always set the option in CACHE

ahoarau committed:
no need for PROJECT_NAME using options

ahoarau committed:
no need for PROJECT_NAME using options

ahoarau committed:
for options to be initialized to boolean type

ahoarau committed:
for options to be initialized to boolean type

ahoarau committed:
make FALLBACK mandatory

ahoarau committed:
make FALLBACK mandatory

ahoarau committed:
jrl_find_package: remove unused feature MODULE_PATH

ahoarau committed:
jrl_find_package: remove unused feature MODULE_PATH

ahoarau committed:
_jrl_check_no_unrecognized_arguments: enforce no unrecognized arguments

ahoarau committed:
_jrl_check_no_unrecognized_arguments: enforce no unrecognized arguments

ahoarau committed:
doc: update api documentation

ahoarau committed:
doc: update api documentation

ahoarau committed:
add release script to check and set versions

ahoarau committed:
add release script to check and set versions

ahoarau committed:
add qol actions

ahoarau committed:
add qol actions

ahoarau committed:
add documentation

ahoarau committed:
add documentation

ahoarau committed:
handle case when no version files are found

ahoarau committed:
handle case when no version files are found

ahoarau committed:
refactor: update version bump commands and improve output options
like uv https://github.com/astral-sh/uv/commit/f401d9ba8feba1ab8a174f5eacf973b7b4b02922

ahoarau committed:
refactor: update version bump commands and improve output options
like uv https://github.com/astral-sh/uv/commit/f401d9ba8feba1ab8a174f5eacf973b7b4b02922

ahoarau committed:
add release.py unit tests and readme

ahoarau committed:
add release.py unit tests and readme

ahoarau committed:
add pixi support

ahoarau committed:
add pixi support

ahoarau committed:
display more user friendly message

ahoarau committed:
display more user friendly message

ahoarau committed:
add back unit tests

ahoarau committed:
add back unit tests

ahoarau committed:
add v2 ci

ahoarau committed:
add v2 ci

ahoarau committed:
move pixi.toml to top level

ahoarau committed:
move pixi.toml to top level

ahoarau committed:
install catch2 in sustem

ahoarau committed:
install catch2 in sustem

ahoarau committed:
use fetchcontent source dir

ahoarau committed:
use fetchcontent source dir

ahoarau committed:
add verbosity

ahoarau committed:
add verbosity

ahoarau committed:
fix tests

ahoarau committed:
fix tests

ahoarau committed:
allow to cancel ci group

ahoarau committed:
allow to cancel ci group

ahoarau committed:
install catch2 in /usr/local

ahoarau committed:
install catch2 in /usr/local

ahoarau committed:
try to use github cache instead of registry

ahoarau committed:
try to use github cache instead of registry

ahoarau committed:
update repo name in test

ahoarau committed:
update repo name in test

ahoarau committed:
use ref to vcpkg cache key

ahoarau committed:
use ref to vcpkg cache key

ahoarau committed:
pull and push vcpkg even if the whole ci fails

ahoarau committed:
pull and push vcpkg even if the whole ci fails

ahoarau committed:
add debug log level in windows ci

ahoarau committed:
add debug log level in windows ci

ahoarau committed:
use CMAKE_TOOLCHAIN_FILE as env var

ahoarau committed:
use CMAKE_TOOLCHAIN_FILE as env var

ahoarau committed:
disable pixi build for now (windows error on symlink)

ahoarau committed:
disable pixi build for now (windows error on symlink)

ahoarau committed:
fix pixi build

ahoarau committed:
fix pixi build

ahoarau committed:
migrate jrl to v2 api

ahoarau committed:
migrate jrl to v2 api

ahoarau committed:
update ci to new jrl v2

ahoarau committed:
update ci to new jrl v2

ahoarau committed:
add ci to generate and check api

ahoarau committed:
add ci to generate and check api

ahoarau committed:
default tests and docgen to off

ahoarau committed:
default tests and docgen to off

ahoarau committed:
update the jrl v2 options

ahoarau committed:
update the jrl v2 options

ahoarau committed:
refactor pixi file to separate clearly the envs

ahoarau committed:
refactor pixi file to separate clearly the envs

ahoarau committed:
_jrl_generate_api_doc: only write if content is unchanged and notify user

ahoarau committed:
_jrl_generate_api_doc: only write if content is unchanged and notify user

ahoarau committed:
add the jrl_print_banner() by default (temp)

ahoarau committed:
add the jrl_print_banner() by default (temp)

ahoarau committed:
add ros2 ci

ahoarau committed:
add ros2 ci

ahoarau committed:
ci: disable ros2 tests (tmp)

ahoarau committed:
ci: disable ros2 tests (tmp)

ahoarau committed:
jrl_python_compute_install_dir: add cmeel exception
This is a temporary fix for windows until https://github.com/cmake-wheel/cmeel/issues/252

ahoarau committed:
jrl_python_compute_install_dir: add cmeel exception
This is a temporary fix for windows until https://github.com/cmake-wheel/cmeel/issues/252

ahoarau committed:
do not expand the Version Changed Panel

ahoarau committed:
do not expand the Version Changed Panel

ahoarau committed:
release.py: handle git and precommit gracefully

ahoarau committed:
release.py: handle git and precommit gracefully

ahoarau committed:
release.py: display git command and only commit the files changed by the version upgrade

ahoarau committed:
release.py: display git command and only commit the files changed by the version upgrade

ahoarau committed:
release.py: allow to detect multiline project()

ahoarau committed:
release.py: allow to detect multiline project()

ahoarau committed:
release.py: use cmake parser to parser cmakelists files

ahoarau committed:
release.py: use cmake parser to parser cmakelists files

ahoarau committed:
release.py: improve the version diff display

ahoarau committed:
release.py: improve the version diff display

ahoarau committed:
release.py: handle user CTRL+C

ahoarau committed:
release.py: handle user CTRL+C

ahoarau committed:
release.py: display the type of change in the Panel

ahoarau committed:
release.py: display the type of change in the Panel

ahoarau committed:
release.py: improve colors

ahoarau committed:
release.py: improve colors

ahoarau committed:
release.py: add support for updating pixi.lock file

ahoarau committed:
release.py: add support for updating pixi.lock file

ahoarau committed:
release.py: add backup and restore

ahoarau committed:
release.py: add backup and restore

ahoarau committed:
release.py: add custom commit and tag messages

ahoarau committed:
release.py: add custom commit and tag messages

ahoarau committed:
chore: bump version to 1.2.0

ahoarau committed:
chore: bump version to 1.2.0

ahoarau committed:
cmake: update python and nanobind min required

ahoarau committed:
cmake: update python and nanobind min required

ahoarau committed:
cmake: add ros2 package support

ahoarau committed:
cmake: add ros2 package support

ahoarau committed:
ci: on pixi only keep 1 ubuntu and 1 windows

ahoarau committed:
ci: on pixi only keep 1 ubuntu and 1 windows

ahoarau committed:
jrl_target_generate_config_header: use arg_UNPARSED_ARGUMENTS to be able to pass VERSION

ahoarau committed:
jrl_target_generate_config_header: use arg_UNPARSED_ARGUMENTS to be able to pass VERSION

nim65s commented on v2/modules/jrl.cmake:
```diff
@@ -0,0 +1,4299 @@
+    execute_process(
+        COMMAND
+            ${python} -c
+            "import compileall; compileall.compile_dir(r'${arg_DIRECTORY}', workers=0, quiet=${quiet_flag})"
```
```suggestion
            ${python} -m compileall ${arg_DIRECTORY}' -j 0 ${quiet_flag}
```

ahoarau committed:
jrl_python_compile_all: use module directly

ahoarau committed:
jrl_python_compile_all: use module directly

ahoarau committed:
v2: fix install fix to v2 folder

ahoarau committed:
v2: fix install fix to v2 folder

ahoarau committed:
gitignore: ignore conda files

ahoarau committed:
gitignore: ignore conda files

ahoarau committed:
tests: make sure the v2 module is present

ahoarau committed:
tests: make sure the v2 module is present

ahoarau committed:
v2: add back the jrl-cmakemodules::jrl-cmakemodules component for backward compat

ahoarau committed:
v2: add back the jrl-cmakemodules::jrl-cmakemodules component for backward compat

ahoarau committed:
pixi: update lock file

ahoarau committed:
pixi: update lock file

ahoarau committed:
v2: use the component template (default)

ahoarau committed:
v2: use the component template (default)

ahoarau committed:
add project tests for find_package on v1 and v2

ahoarau committed:
add project tests for find_package on v1 and v2

ahoarau committed:
tests: include base before project()

ahoarau committed:
tests: include base before project()

ahoarau committed:
jrl_configure_default_install_dirs: prevent warning on pure-cmake project

ahoarau committed:
jrl_configure_default_install_dirs: prevent warning on pure-cmake project

ahoarau committed:
jrl_export_package: loop over each target to install headers

ahoarau committed:
jrl_export_package: loop over each target to install headers

ahoarau committed:
tests: refactor current testsuite and add multitargets test

ahoarau committed:
tests: refactor current testsuite and add multitargets test

ahoarau committed:
tests: add multi component test project

ahoarau committed:
tests: add multi component test project

ahoarau committed:
test: remove "" for cmake 3.22

ahoarau committed:
test: remove "" for cmake 3.22

ahoarau committed:
pixi: add cmake322 test env

ahoarau committed:
pixi: add cmake322 test env

ahoarau committed:
pixi: update lock file

ahoarau committed:
pixi: update lock file

ahoarau committed:
ci: test cmake322 env on all platforms

ahoarau committed:
ci: test cmake322 env on all platforms

ahoarau committed:
tests: add minimal project with uppercase in the name

ahoarau committed:
tests: add minimal project with uppercase in the name

ahoarau committed:
v2: suport UpperCase project names via Config.cmake and ConfigVersion.cmake

ahoarau committed:
v2: suport UpperCase project names via Config.cmake and ConfigVersion.cmake

ahoarau committed:
doc: update docs

ahoarau committed:
doc: update docs

ahoarau committed:
ci: add explicit test-pixi-build env

ahoarau committed:
ci: add explicit test-pixi-build env

ahoarau committed:
tests: require jrl version 1.2.0 min

ahoarau committed:
tests: require jrl version 1.2.0 min

ahoarau committed:
pixi: update lock file

ahoarau committed:
pixi: update lock file

ahoarau committed:
remove unused generated readme

ahoarau committed:
remove unused generated readme

ahoarau committed:
pixi: integrate python tests

ahoarau committed:
pixi: integrate python tests

ahoarau committed:
pixi: update lock file

ahoarau committed:
pixi: update lock file

ahoarau committed:
release.py: remove gitpython dep and cleanup comments

ahoarau committed:
release.py: remove gitpython dep and cleanup comments

ahoarau committed:
release.py: update/shorten doc

ahoarau committed:
release.py: update/shorten doc

ahoarau committed:
release.py: remove enumerate

ahoarau committed:
release.py: remove enumerate

ahoarau committed:
release.py: change ChangelogVersionExtractor to inherit from VersionExtractor

ahoarau committed:
release.py: change ChangelogVersionExtractor to inherit from VersionExtractor

ahoarau committed:
release.py: Add option to specify files to stage

ahoarau committed:
release.py: Add option to specify files to stage

ahoarau committed:
release.py: display git tag and commit in dry run

ahoarau committed:
release.py: display git tag and commit in dry run

ahoarau committed:
refactor: simplify update_pixi_lock function params

ahoarau committed:
refactor: simplify update_pixi_lock function params

ahoarau committed:
test: update assertion for previous_version in cli json output test

ahoarau committed:
test: update assertion for previous_version in cli json output test

ahoarau committed:
tests: add release.py coverage

ahoarau committed:
tests: add release.py coverage

ahoarau committed:
release.py: update dry run output message

ahoarau committed:
release.py: update dry run output message

ahoarau committed:
pixi: add doc task to generate api doc

ahoarau committed:
pixi: add doc task to generate api doc

ahoarau committed:
doc: update documentation

ahoarau committed:
doc: update documentation

ahoarau committed:
readme: update main readme for v2

ahoarau committed:
readme: update main readme for v2

ahoarau committed:
release.py: display the output of pixi list command

ahoarau committed:
release.py: display the output of pixi list command

ahoarau committed:
release.py: uniformize the display output with the dry-run

ahoarau committed:
release.py: uniformize the display output with the dry-run

ahoarau committed:
rename release.py to jrl_release.py

ahoarau committed:
rename release.py to jrl_release.py

ahoarau committed:
jrl_release.py: update sheban according to ub doc and chmod +x

ahoarau committed:
jrl_release.py: update sheban according to ub doc and chmod +x

ahoarau committed:
jrl_release.py: remove unused RegexVersionExtractor

ahoarau committed:
jrl_release.py: remove unused RegexVersionExtractor

ahoarau committed:
tests: remove test on RegexVersionExtractor
which has been removed because unused

ahoarau committed:
tests: remove test on RegexVersionExtractor
which has been removed because unused

ahoarau committed:
jrl_target_set_default_compile_options: update doc

ahoarau committed:
jrl_target_set_default_compile_options: update doc

ahoarau committed:
jrl_target_enforce_msvc_conformance: update docs

ahoarau committed:
jrl_target_enforce_msvc_conformance: update docs

ahoarau committed:
ci: add boost-libs (required on boost 1.90)

ahoarau committed:
ci: add boost-libs (required on boost 1.90)

ahoarau committed:
cmake: reformat with new gersemi

ahoarau committed:
cmake: reformat with new gersemi

ahoarau committed:
jrl: update error messages

ahoarau committed:
jrl: update error messages

ahoarau committed:
pre-commit: update gersemi new repo

ahoarau committed:
pre-commit: update gersemi new repo

ahoarau committed:
gitignore: merge v2 with main

ahoarau committed:
gitignore: merge v2 with main

ahoarau committed:
tests: add missing multicomp-consumer-all test

ahoarau committed:
tests: add missing multicomp-consumer-all test

ahoarau committed:
tests: add multi comp + generator expression test

ahoarau committed:
tests: add multi comp + generator expression test

ahoarau committed:
_jrl_export_dependencies: export all deps to support generator expressions

ahoarau committed:
_jrl_export_dependencies: export all deps to support generator expressions

ahoarau committed:
pixi: add eigen to the test deps

ahoarau committed:
pixi: add eigen to the test deps

ahoarau committed:
pixi: update lock file

ahoarau committed:
pixi: update lock file

ahoarau committed:
generate-dependencies: handle the missing json case

ahoarau committed:
generate-dependencies: handle the missing json case

ahoarau committed:
tests: add regular find_package test

ahoarau committed:
tests: add regular find_package test

ahoarau committed:
ci: add eigen dep to tests

ahoarau committed:
ci: add eigen dep to tests

ahoarau committed:
jrl: remove unused jrl_print_full_system_and_compiler_info

ahoarau committed:
jrl: remove unused jrl_print_full_system_and_compiler_info

ahoarau committed:
pre-commit: update dependencies

ahoarau committed:
pre-commit: update dependencies

ahoarau committed:
pixi: update tests dependencies

ahoarau committed:
pixi: update tests dependencies

ahoarau committed:
pixi: update lock file

ahoarau committed:
pixi: update lock file

ahoarau committed:
package.xml: update tests dependencies

ahoarau committed:
package.xml: update tests dependencies

ahoarau committed:
cmake: add status message when building tests

ahoarau committed:
cmake: add status message when building tests

ahoarau committed:
jrl_configure_default_build_type: log build type

ahoarau committed:
jrl_configure_default_build_type: log build type

ahoarau committed:
ci-ros: use master tag as recommended

ahoarau committed:
ci-ros: use master tag as recommended

ahoarau committed:
ci: update tests dependencies

ahoarau committed:
ci: update tests dependencies

ahoarau committed:
jrl_python_get_interpreter: move it up so jrl_find_python can use it

ahoarau committed:
jrl_python_get_interpreter: move it up so jrl_find_python can use it

ahoarau committed:
find-modules: handle QUIET mode

ahoarau committed:
find-modules: handle QUIET mode

ahoarau committed:
jrl_find_python: show the Python::Interpreter location

ahoarau committed:
jrl_find_python: show the Python::Interpreter location

ahoarau committed:
_jrl_export_dependencies: filter out the project's exported targets during the dependency extration

ahoarau committed:
_jrl_export_dependencies: filter out the project's exported targets during the dependency extration

ahoarau committed:
tests: refactor tests with pixi and lib/consumer approach

ahoarau committed:
tests: refactor tests with pixi and lib/consumer approach

ahoarau committed:
tests: fix tests with pixi

ahoarau committed:
tests: fix tests with pixi

ahoarau committed:
ci: add tests with pixi

ahoarau committed:
ci: add tests with pixi

ahoarau committed:
tests: add minimum CMAKE_OSX_DEPLOYMENT_TARGET 11

ahoarau committed:
tests: add minimum CMAKE_OSX_DEPLOYMENT_TARGET 11

ahoarau committed:
ci: use Python_ROOT to guide the search

ahoarau committed:
ci: use Python_ROOT to guide the search

ahoarau committed:
tests: remove the jrl dep in find-modules tests

ahoarau committed:
tests: remove the jrl dep in find-modules tests

ahoarau committed:
ci: add debug info

ahoarau committed:
ci: add debug info

ahoarau committed:
tests: shorter minimal proj

ahoarau committed:
tests: shorter minimal proj

ahoarau committed:
tests: shorten lib names

ahoarau committed:
tests: shorten lib names

ahoarau committed:
tests: fix tests names

ahoarau committed:
tests: fix tests names

ahoarau committed:
pixi: add cross intel build

ahoarau committed:
pixi: add cross intel build

ahoarau committed:
ci: put vcpkg pip to path

ahoarau committed:
ci: put vcpkg pip to path

ahoarau committed:
ci: add python path to env

ahoarau committed:
ci: add python path to env

ahoarau committed:
ci: use global env for ctest

ahoarau committed:
ci: use global env for ctest

ahoarau committed:
ci: add python debug info

ahoarau committed:
ci: add python debug info

ahoarau committed:
ci: disable pixi test on windows

ahoarau committed:
ci: disable pixi test on windows

ahoarau committed:
tests: add a simple test on header export

ahoarau committed:
tests: add a simple test on header export

ahoarau committed:
_jrl_check_dir_exists: allow custom error message

ahoarau committed:
_jrl_check_dir_exists: allow custom error message

ahoarau committed:
jrl_cmakemodules_get_version: fix error message

ahoarau committed:
jrl_cmakemodules_get_version: fix error message

ahoarau committed:
jrl_configure_default_build_type: fix wrong variable name

ahoarau committed:
jrl_configure_default_build_type: fix wrong variable name

ahoarau committed:
jrl: add missing checks

ahoarau committed:
jrl: add missing checks

ahoarau committed:
jrl_target_headers: do not install PRIVATE headers

ahoarau committed:
jrl_target_headers: do not install PRIVATE headers

ahoarau committed:
tests: add test for jrl_target_headers visibility

ahoarau committed:
tests: add test for jrl_target_headers visibility

ahoarau committed:
_jrl_normalize_version: fix doc

ahoarau committed:
_jrl_normalize_version: fix doc

ahoarau committed:
_jrl_target_generate_header: remove unused INSTALL_DESTINATION

ahoarau committed:
_jrl_target_generate_header: remove unused INSTALL_DESTINATION

ahoarau committed:
jrl: update doc

ahoarau committed:
jrl: update doc

ahoarau committed:
doc: regenerate api doc

ahoarau committed:
doc: regenerate api doc

ahoarau committed:
tests: remove build and instal before runnig ctest

ahoarau committed:
tests: remove build and instal before runnig ctest

ahoarau committed:
jrl: add _jrl_check_var_not_defined and _jrl_check_strequal

ahoarau committed:
jrl: add _jrl_check_var_not_defined and _jrl_check_strequal

ahoarau committed:
jrl_test_case: new function to declare pure cmake tests via add_test

ahoarau committed:
jrl_test_case: new function to declare pure cmake tests via add_test

ahoarau committed:
tests: add more unit tests in pure cmake

ahoarau committed:
tests: add more unit tests in pure cmake

ahoarau committed:
tests: refacor test with new test framework

ahoarau committed:
tests: refacor test with new test framework

ahoarau committed:
jrl: add new function _jrl_check

ahoarau committed:
jrl: add new function _jrl_check

ahoarau committed:
jrl: replace some _jrl_check_* with _jrl_check()

ahoarau committed:
jrl: replace some _jrl_check_* with _jrl_check()

ahoarau committed:
tests: refactor tests with new jrl_check()

ahoarau committed:
tests: refactor tests with new jrl_check()

ahoarau committed:
_jrl_init_dirs: replace the _jrl_<>_dir functions by variables for direct call

ahoarau committed:
_jrl_init_dirs: replace the _jrl_<>_dir functions by variables for direct call

ahoarau committed:
jrl: add back shortcuts for api clarity

ahoarau committed:
jrl: add back shortcuts for api clarity

ahoarau committed:
jrl_release: treat version not present in pyproject.toml as warning

ahoarau committed:
jrl_release: treat version not present in pyproject.toml as warning

ahoarau committed:
precommit: upgrade versions

jorisv commented on .github/workflows/v2-linux.yml:
```diff
@@ -0,0 +1,123 @@
+          cmake-version: "3.22"
+      - uses: actions/setup-python@v6
+        with:
+          python-version: "3.10"
```
It's better to directly run on the Ubuntu docker image (`'ubuntu:22.04'` and 'ubuntu:24.04'`).
As you noticed, the Github Ubuntu image is heavily modified. Using the docker image ensure you're testing on the right software stack and avoid using too much github action (minimizing maintenance time).

jorisv commented on v2/tests/find_modules/FindAccelerateQuiet/CMakeLists.txt:
```diff
@@ -0,0 +1,25 @@
+    return()
+endif()
+
+# Restrict search paths to ensure Pytest is not found on the system.
```
```suggestion
# Restrict search paths to ensure Accelerate is not found on the system.
```

jorisv commented on v2/tests/find_modules/FindAccelerateNotFound/CMakeLists.txt:
```diff
@@ -0,0 +1,25 @@
+cmake_minimum_required(VERSION 3.22)
+project(FindAccelerateQuietTest)
+
```
I put this comment here, but it apply on all Quiet tests.

Quiet version test if the finder fail when the dependency is not installed. The naming is not clear, you can rename it to Fail or NotFound (FindmatioNotFound or FindmatioFail)

jorisv commented on v2/tests/find_modules/FindmatioQuiet/CMakeLists.txt:
```diff
@@ -0,0 +1,38 @@
+
+set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/../../../find-modules")
+
+# Restrict search paths to ensure Pytest is not found on the system.
```
```suggestion
# Restrict search paths to ensure matio is not found on the system.
```

jorisv commented on v2/tests/find_modules/FindsimdeQuiet/CMakeLists.txt:
```diff
@@ -0,0 +1,31 @@
+
+set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/../../../find-modules")
+
+# Restrict search paths to ensure Pytest is not found on the system.
```
```suggestion
# Restrict search paths to ensure simde is not found on the system.
```

jorisv commented on v2/tests/CMakeLists.txt:
Test naming is not clear (hdr, mt, mc, mcg, rfp ?).
Add a README.md to explain each test purpose and responsibility.

ahoarau replied:
Done and refactored/cleaned up the tests.

jorisv commented on v2/tests/rfp/CMakeLists.txt:
```diff
@@ -0,0 +1,19 @@
+target_link_libraries(t INTERFACE Eigen3::Eigen)
+
+jrl_add_export_component(NAME t TARGETS t)
+jrl_export_package()
```
As far as I understand, nothing is tested here. This test only produce a warning.
It can be tested with a consumer failing to import the generated cmake module.

jorisv replied:
After discussion this test jrl_add_export_component and jrl_export_package doesn't fail.

jorisv commented on v2/tests/.gitignore:
```diff
@@ -0,0 +1,4 @@
+.pixi
+build*/
+install*/
+pixi.lock
```
Instead of adding a .gitignore in all subdirectory it's possible to use a glob pattern :

> A leading "**" followed by a slash means match in all directories. For example, "**/foo" matches file or directory "foo" anywhere, the same as pattern "foo". "**/foo/bar" matches file or directory "bar" anywhere that is directly under directory "foo".

```suggestion
**/.pixi
**/build*/
**/install*/
**/pixi.lock
```

jorisv commented on v2/tests/modules/test_jrl_export_dependency.cmake:
```diff
@@ -0,0 +1,155 @@
+    jrl_add_export_component(NAME comp_y TARGETS shared_target)
+  ]]
+  WILL_FAIL
+)
```
`jrl_add_export_compenent` is already tested in `test_jrl_add_export_component.cmake`

jorisv commented on v2/tests/no_deps/CMakeLists.txt:
What the purpose of this test ?

It install a library without any dependency. Is there a special edge case to test compared to a library with one dependency ? Especially compared to `MinimalProj`.

ahoarau replied:
yes there's a special handling when the project does not declare any dependencies (generate-dependencies.cmake must still "work"). This must stay to lock the behavior.
FYI: The tests projects have been added on the go to test/highlight a particular feature. But as they are integration tests, some overlap is unavoidable, but that's fine. The more there are the better.

jorisv commented on v2/tests/pybindings/pb-library/tests/CMakeLists.txt:
```diff
@@ -0,0 +1,26 @@
+add_test(
+    NAME ${test_name}
+    COMMAND $<TARGET_FILE:Pytest::Pytest> ${CMAKE_CURRENT_SOURCE_DIR}/test_nanobind_bindings.py
+)
```
One use case of Pytest is to auto-discover tests. Is there a reason to not use this feature ?

ahoarau replied:
I want to have 2 lines showed in the ctest (1 for nanobind, 1 for boost python). 
But of course, users can use whatever.

jorisv commented on v2/tests/pybindings/pb-library/tests/CMakeLists.txt:
```diff
@@ -0,0 +1,26 @@
+    ${test_name}
+    PROPERTIES
+        ENVIRONMENT_MODIFICATION
+            "PYTHONPATH=path_list_prepend:${CMAKE_BINARY_DIR}/lib/site-packages;PYTHONPATH=path_list_prepend:${CMAKE_CURRENT_BINARY_DIR}"
```
Why adding `PYTHONPATH=path_list_prepend:${CMAKE_CURRENT_BINARY_DIR}`  ?
It's not done with boost python test.

ahoarau replied:
uniformized 👍

jorisv commented on v2/tests/pybindings/pb-library/tests/CMakeLists.txt:
```diff
@@ -0,0 +1,26 @@
+
+catch_discover_tests(test_StringUtils)
+
+set(test_name "Nanobind Bindings")
```
Python tests are added even if `PB_BUILD_PYTHON_BINDINGS` is false.

jorisv commented on v2/tests/pybindings/pb-library/pixi.toml:
```diff
@@ -0,0 +1,53 @@
+configure = { cmd = "cmake --log-level=DEBUG -G Ninja -S . -B build-intel" }
+build = { cmd = "cmake --build build-intel --verbose" }
+install = { cmd = "cmake --install build-intel" }
+test = { depends-on = ["clear", "configure", "build", "install"] }
```
Why a separate feature for osx-64 ?

ahoarau replied:
Just an old cross-compile feature for macos. Removed.

jorisv commented on README.md:
```diff
@@ -1,16 +1,23 @@
 at CNRS, LAAS/HPP, JRL and Inria.
 
 It factorizes CMake mechanisms to provide a uniform look'n feel for
 all packages.
```
```suggestion
jrl-cmakemodules provide functions to factorize and write more reliable CMake code.

It's used by INRIA and CNRS teams but can easily be integrated in any project.
```

jorisv commented on CMakeLists.txt:
Add an option to only install V2. This will avoid license issue with conda package.

ahoarau replied:
Added `JRL_CMAKEMODULES_INSTALL_V2_ONLY` Cmake option (ON default) + `pixi run install_v2`

jorisv commented on v2/modules/jrl.cmake:
When running ccmake or cmake-gui a lot of entry are visible:
- CMAKE_ARCHIVE_OUTPUT_DIRECTORY{,_DEBUG,_MINSIZEREL,_RELEASE,_RELWITHDEBINFO}
- CMAKE_LIRBRARY_OUTPUT_DIRECTORY{,_DEBUG,_MINSIZEREL,_RELEASE,_RELWITHDEBINFO}
- CMAKE_RUNTIME_OUTPUT_DIRECTORY{,_DEBUG,_MINSIZEREL,_RELEASE,_RELWITHDEBINFO}

Is it possible to mark them ADVENCED ?

jorisv commented on v2/find-modules/Findmatio.cmake:
We need FindCHOLMOD.cmake that is not packaged on Ubuntu 22.04 `libsuitesparse-dev` package.
It's only packaged on Ubuntu 24.04.

ahoarau replied:
Done, added from https://github.com/ahoarau/FindCHOLMOD.cmake

jorisv commented on v2/modules/jrl.cmake:
`jrl_find_package` doesn't support workspace.

When creating the following root `CMakeLists.txt`:
```cmake
cmake_minimum_required(VERSION 3.22)
project(root)
include(CTest)

add_subdirectory(eigenpy-jrl-next)
add_subdirectory(coal-jrl-next)
```

With the following `pixi.toml`:
```toml
[workspace]
name = "workspace"
version = "1.0.0"
platforms = ["linux-64"]
channels = ["conda-forge", "../../feedstock/staged-recipes/build_artifacts"]
preview = ["pixi-build"]

[dependencies]
ccache = ">=4.9.1"
cmake = ">=3.10"
cxx-compiler = ">=1.7.0"
ninja = ">=1.11"
pkg-config = ">=0.29.2"
git = ">=2.47.0"
libboost-devel = ">=1.80.0"
libboost-python-devel = ">=1.80.0"
eigen = ">=3.4.0"
numpy = ">=1.22.0"
python = ">=3.10"
scipy = ">=1.10.0"
doxygen = ">=1.13.2,<2"
lxml = ">=5.3"
pylatexenc = ">=2.10"
assimp = ">=5.4"
graphviz = ">=13.1.2,<14"
octomap = ">=1.10"
qhull = ">=2020.2"
qhull-static = ">=2020.2"
suitesparse = ">=5"
jrl-cmakemodules = "*"

[target.unix.activation]
scripts = ["eigenpy-jrl-next/development/scripts/pixi/activation.sh"]

[activation.env]
CMAKE_CXX_COMPILER_LAUNCHER = "ccache"
CMAKE_EXPORT_COMPILE_COMMANDS = "ON"
CMAKE_COLOR_DIAGNOSTICS = "ON"
CCACHE_SLOPPINESS = "include_file_ctime,include_file_mtime,pch_defines,time_macros"

[tasks]
configure = { cmd = [
  "CXXFLAGS=$EIGENPY_CXX_FLAGS",
  "cmake",
  "-G",
  "Ninja",
  "-B",
  "build",
  "-S",
  ".",
  "-DCMAKE_INSTALL_PREFIX=$CONDA_PREFIX",
  "-DCMAKE_BUILD_TYPE=$RELEASE",
  "-DBUILD_TESTING=ON",
  "-DBUILD_TESTING_SCIPY=ON",
  "-DGENERATE_PYTHON_STUBS=ON",
  "-DBUILD_WITH_CHOLMOD_SUPPORT=ON",
  "-DBUILD_WITH_ACCELERATE_SUPPORT=OFF",
  "-DCOAL_BUILD_WITH_OCTOMAP=ON",
  "-DCOAL_BUILD_WITH_QHULL=ON",
  "-DBUILD_PYTHON_INTERFACE=ON",
] }
```

Running `pixi run configure` give the following error message:
```
-- [eigenpy]
--    Executing find_package(eigenpy CONFIG REQUIRED)
CMake Error at .pixi/envs/default/share/cmake/jrl-cmakemodules/v2/modules/jrl.cmake:1815 (find_package):
  Could not find a package configuration file provided by "eigenpy" with any
  of the following names:

    eigenpy.cps
    eigenpyConfig.cmake
    eigenpy-config.cmake

  Add the installation prefix of "eigenpy" to CMAKE_PREFIX_PATH or set
  "eigenpy_DIR" to a directory containing one of the above files.  If
  "eigenpy" provides a separate development package or SDK, be sure it has
  been installed.
Call Stack (most recent call first):
  coal-jrl-next/CMakeLists.txt:147 (jrl_find_package)
```

To overcome this issue, we must apply the following patch to coal `CMakeLists.txt`:

```patch
diff --git a/CMakeLists.txt b/CMakeLists.txt
index a210c3ee..9dfee39c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -144,7 +144,9 @@ if(BUILD_PYTHON_INTERFACE AND COAL_PYTHON_NANOBIND)
 endif()
 
 if(BUILD_PYTHON_INTERFACE AND NOT COAL_PYTHON_NANOBIND)
-  jrl_find_package(eigenpy CONFIG REQUIRED)
+  if(NOT TARGET eigenpy::eigenpy)
+    jrl_find_package(eigenpy CONFIG REQUIRED)
+  endif()
 endif()
 
 if(COAL_BUILD_DOCUMENTATION)
-- 
```

This use case was managed by the `PROJECT_PACKAGES_IN_WORKSPACE` variable in current jrl-cmakemodules.
At some point we were speaking about adding target as `jrl_find_package` argument to manage this issue.

ahoarau replied:
As discussed, I will add back the `EXPECTED_TARGETS` to `jrl_find_package`. 
Not great, but fair enough.

ahoarau replied:
Done in 5ae105c0bec184529f61f80e97088c25c5144945 👍

jorisv commented on v2/scripts/jrl_release.py:
It's working but a little hard to use. User must know where jrl-cmakemodule is installed to run this script.
I would advocate to turn this to a real python project (jrl-cmakemodules-scripts) with a dedicated pyproject.toml. So we can package it in nix/conda/pypi and use it easily.

ahoarau replied:
pyproject.toml added, with a readme section. Then you can access the tool via `jrl-release --check-version`.

ahoarau committed:
tests: fix typo in comment

ahoarau committed:
gitignore: gather gitignore in tests

ahoarau committed:
tests: remove duplicated tests for jrl_add_export_component in jrl_export_dependency file

ahoarau committed:
jrl_find_package: add EXPECTED_TARGETS keyword to unlock superbuilds
Also add unit test to lock the behavior.

ahoarau committed:
test: relax cxx-compiler constraint

ahoarau committed:
pixi: update lock file

ahoarau committed:
tests: rename Quiet tests to NotFound

ahoarau committed:
jrl_configure_default_binary_dirs: mark as advanced

ahoarau committed:
doxygen.cmake: add working directory when installing

ahoarau committed:
tests: fix path

ahoarau committed:
ci: use ubuntu docker images instead of github images

ahoarau committed:
jrl_export_package: remove one extra LR

ahoarau committed:
pixi: remove cmake322 + pixi-build env
cmake322 is creating issues on macos26. It forces. clang18 which is no longer compatible with the latest Xcode SDK.
cmake 3.22 is tested on ubuntu 22.04.

Pixi build env is removed as it is used extensively in tests.

ahoarau committed:
tests: remove cross compile feature on mac

ahoarau committed:
tests: run python tests only if PB_BUILD_PYTHON_BINDINGS is ON

ahoarau committed:
test: pb-library uniformize binding tests with pytest

ahoarau committed:
tests: refactor v2 tests
* renamed folders and structure for coherence
* "pixi run test" works on all levels
* added the CamelCase test
* remove duplicated tests

ahoarau committed:
tests: define test_with_pixi pixi task

ahoarau committed:
jrl: add JRL_CMAKEMODULES_INSTALL_V2_ONLY option to install only v2 + pixi tasks

ahoarau committed:
readme: update project description

ahoarau committed:
v2: add FindCHOLMOD.cmake for ubuntu 22.04 that ship an old suitesparse without CHOLMODConfig.cmake

ahoarau committed:
tests: add FindCHOLMOD.cmake unit test

ahoarau committed:
apply dos2unix

ahoarau committed:
add missing suitesparse dep

ahoarau committed:
pixi: update lock file

ahoarau committed:
add pyproject.toml to allow jrl-release.py packaging

nim65s commented on pyproject.toml:
```diff
@@ -0,0 +1,28 @@
+name = "jrl-cmakemodules-scripts"
+version = "1.2.0"
+description = "Release scripting tools for JRL CMake modules"
+readme = "v2/scripts/README.md"
```
this file is missing

ahoarau replied:
this file is now there !

ahoarau committed:
jrl_release: update deps

ahoarau committed:
jrl_release: add README

ahoarau committed:
pyproject.toml: allow to run tests

ahoarau committed:
ci: add python scripts ci

ahoarau committed:
gitignore: ignore uv.lock

ahoarau committed:
jrl_release: ensure utf-8 console (windows fix)

ahoarau committed:
tests: remove the redundant exported_header test
header_visibility is already covering those cases

ahoarau committed:
remove extra gitignore

ahoarau committed:
tests: fix the regular_find_package main pixi test task

ahoarau committed:
jrl_print_dependencies_summary: handle in-tree dependencies

ahoarau committed:
tests: add missing pixi deps

ahoarau committed:
jrl: change verbosity and remove symbols for [OK] and [NOT_FOUND]

nim65s commented on package.xml:
```diff
@@ -106,10 +106,23 @@
   -->
 
   <buildtool_depend>cmake</buildtool_depend>
+  <buildtool_depend>ninja-build</buildtool_depend>
```
wait what ?

ahoarau replied:
reomved

nim65s commented on pyproject.toml:
```diff
@@ -0,0 +1,43 @@
+]
+
+[project.urls]
+Homepage = "https://github.com/jrl-umi3218/jrl-cmakemodules"
```
```suggestion
homepage = "https://github.com/jrl-umi3218/jrl-cmakemodules"
```

ref https://pypi.org/project/pin/ & https://github.com/cmake-wheel/pinocchio/blob/af222648db07c26f807764a5944a3e82e2d664ca/pyproject.toml#L27

ahoarau committed:
pyproject.toml: fix case

Co-authored-by: Guilhem Saurel <guilhem.saurel@laas.fr>

ahoarau committed:
package.xml: remove ninja-build dep

ahoarau committed:
move pyproject.toml to v2/scripts

ahoarau committed:
tests: add standalone add_cmake_tests.cmake with COMMAND and ENVIRONMENT arg

ahoarau committed:
jrl_print_dependencies_summary: add DEPENDS argument

ahoarau committed:
jrl_export_package: improve doc and allow to specify an env variable to disable export

ahoarau committed:
tests: test the print disabling api

ahoarau committed:
test: add export_package_control test
